jetty-server/src/main/java/org/eclipse/jetty/server/InclusiveByteRange.java

```diff
@@ -100,52 +169,89 @@ public long getLast()
                     try
                     {
                         t = tok.nextToken().trim();
+                        if ("bytes".equals(t))
+                            continue;
 
                         long first = -1;
                         long last = -1;
-                        int d = t.indexOf('-');
-                        if (d < 0 || t.indexOf("-",d + 1) >= 0)
+                        int dash = t.indexOf('-');
+                        if (dash < 0 || t.indexOf("-",dash + 1) >= 0)
                         {
-                            if ("bytes".equals(t))
-                                continue;
                             LOG.warn("Bad range format: {}",t);
-                            continue headers;
+                            break;
                         }
-                        else if (d == 0)
+                        
+                        if (dash>0)
+                            first = Long.parseLong(t.substring(0,dash).trim());
+                        if (dash<(t.length()-1))
+                            last = Long.parseLong(t.substring(dash + 1).trim());
+                        
+                        if (first==-1)
                         {
-                            if (d + 1 < t.length())
-                                last = Long.parseLong(t.substring(d + 1).trim());
-                            else
+                            if (last==-1)
                             {
                                 LOG.warn("Bad range format: {}",t);
```

Comment:
I'm beginning to think that and bad range should result in a 400 Bad Request response.

Reply:
Probably, but let's not change that behaviour in this release.... we'd have to add compliance modes etc.